### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by mohamed20163858**
in this pull request, I added an authentication page to enable anyone to register to the blog app using the `devise` gem, and here are the things I did:- 
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords stored in the database.
- Ask for confirmation by email.
- Can reset password.
so to enable you test it in a proper way you need to write these command `rails db:seed` 
and here are some   screen shots for the app I created:- 

commands needed to initialize the app 
image

put some data inside the database then start the server:- 
image

when you try to navigate you to that URL http://127.0.0.1:3000 it will automatically navigate you to sign in page because you are not authenticated yet 
image

just click to sign up link and fill out this form : -
image

devise will send you confirmation mail including confirmation link and cause it is just in development you will see the mail inside the terminal you executed `rails s ` command and here is a screenshot of the mail 
image
just copy the confirmation link and paste it in your browser tab 
it will redirect you to this page 
image
now click login and you will find yourself in this page 
image

 and  here is a proof the passwords are hashed using devise 
image

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "current-password" %>
+  </div>
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
```
Comment (codepantha) [inline app/views/devise/sessions/new.html.erb:17]: - [x] In keeping with the layout of the wireframe, please ensure that the input fields for the registration and login is modified to match the wireframes and their styling. As you can see, the input fields and submit button are taking up most of the available space on the screen. Please make this change 🙏 
### Wireframe design
image
### Your Design
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @mohamed20163858 👋

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Add authentication with devise 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (mohamed20163858) [reply app/views/devise/sessions/new.html.erb:17]: ok done thanks alot for your feedback

**Review by DammyShittu — APPROVED** [review]
Hi @mohamed20163858 and @Mov305,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**